sdk/python/test_picovoice.py

```diff
@@ -42,34 +40,42 @@ def inference_callback(self, inference):
 class PicovoiceTestCase(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
-        metas = [
-            PicovoiceMetaData('en', 'coffee_maker', ['picovoice']),
-            PicovoiceMetaData('es', 'luz', ['manzana']),
-            PicovoiceMetaData('de', 'beleuchtung', ['heuschrecke'])]
-
-        cls._pvs = list()
-        for meta in metas:
-            keyword_paths = list()
-            for x in meta.keywords:
-                keyword_paths.append(pv_keyword_paths_by_language(meta.language)[x])
-            
-            cls._pvs.append([meta, Picovoice(
+        models = [
+            ('en', 'coffee_maker', 'picovoice'),
+            ('es', 'luz', 'manzana'),
+            ('de', 'beleuchtung', 'heuschrecke')]
+
+        cls._pvs = dict()
+        for model in models:
+            language, context, keyword = model
+            pvTestData = PicovoiceTestData()
+            _pv = Picovoice(
                 access_key=sys.argv[1],
-                keyword_path=keyword_paths[0],
-                context_path=context_path(meta.context, meta.language),
-                porcupine_model_path=pv_porcupine_model_path_by_language(meta.language),
-                rhino_model_path=pv_rhino_model_path_by_language(meta.language),
-                wake_word_callback=meta.wake_word_callback,
-                inference_callback=meta.inference_callback)])
+                keyword_path=pv_keyword_paths_by_language(language)[keyword],
+                context_path=context_path(context, language),
+                porcupine_model_path=pv_porcupine_model_path_by_language(language),
+                rhino_model_path=pv_rhino_model_path_by_language(language),
+                wake_word_callback=pvTestData.wake_word_callback,
+                inference_callback=pvTestData.inference_callback)
+            pvTestData.pv = _pv
+
+            if language not in cls._pvs:
+                cls._pvs[language] = dict()
+            if context not in cls._pvs[language]:
+                cls._pvs[language][context] = dict()
+            cls._pvs[language][context][keyword] = pvTestData
```

Comment:
how about instead of the three-key dictionary you overwrite the `__str__` function of pvTestData as the concatenation of language, context and keyword and use that as the key? Then this would be a single set:
```python
cls._picovoiceDictionary[str(pvTestData)] = pvTestData
```
also check out the [setdefault()](https://www.w3schools.com/python/ref_dictionary_setdefault.asp) function in python. It allows you to set a dictionary without checking if the key exists.

Reply:
actually I thought about keeping a single-key dictionary, I was just worried that concatenation of those 3 strings might not be unique. I guess we can just keep that in mind when we want to design tests (or possibly use a special character in between each string). Also I think it would be easier to do it without overwriting the __str__, since currently I'm not storing language, context or keyword in pvTestData.